### ISSUE TITLE
Apply - to what?

### ISSUE THREAD

**Issue by bumbaras** [Klowner/inkscape-applytransforms#16]
I thought this addon will help me to flatten transformations but i was wrong. It just exchanges one or few transformations add to object into one matrix transformation. I have two apps which i need to import created SVG and they totally don't understand matrix transformation thing. Really don't understand what this addon was for from the very start.

**Comment by Klowner** [comment]
Please provide steps to reproduce.

**Comment by Klowner** [comment]
If you can provide any useful info about how you would like this potential issue to be resolved, or can provide example files and expected results, feel free to reopen the ticket.

**Comment by bumbaras** [comment]
I really don't know what to do to resolve this. I know (atleast now I know) that all modern SVG editors exchange one or many transforms into one matrix transform. It can be efficient as gives shorted file. But it appeared that not every app (not native SVG editor) can be unable to support this. One of example is Schneider TGML editor. There is no way to force them to add such thing, they openly said SVG is only partially supported.
Still applytransform name is a little confusion - it doesn't apply, it creates one big matrix transform.
Such addon as applytransform should in my opinion ... just remove all transforms by modify object into the final stage (recalculate it's points, vertices, coords into fixed values). Don't know if this is even possible.

**Comment by Klowner** [comment]
Could you provide me with an example SVG that you're trying to remove all matrix transforms from?

**Comment by bumbaras** [comment]
I didn't check for response here because was sure there won't be any.
The function to attach files is not working on my end so i can only put a svg code into.
After analyzing applytransform.py code i see You are using functions from simpletransform.py. It clearifies transforms from polygones for example but for rects, and ellipses it leaves me with matrices. How beautiful matrix is and may be, some tools are don't understand matrices at all.
It would be hard to remove matrices from rects because it uses X,Y and width and height and there is no rotate thing in attributes (or maybe there is but i don't know). Rotate is acceptable though.

<?xml version="1.0" encoding="UTF-8" standalone="no"?>


<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   version="1.1"
   id="elementy"
   x="0px"
   y="0px"
   width="1375px"
   height="602px"
   viewBox="0 0 1375 602"
   enable-background="new 0 0 1375 602"
   xml:space="preserve"
   sodipodi:docname="rects.svg"
   inkscape:version="0.92.3 (2405546, 2018-03-11)"><sodipodi:namedview
   pagecolor="#ffffff"
   bordercolor="#666666"
   borderopacity="1"
   objecttolerance="10"
   gridtolerance="10"
   guidetolerance="10"
   inkscape:pageopacity="0"
   inkscape:pageshadow="2"
   inkscape:window-width="1280"
   inkscape:window-height="962"
   id="namedview471"
   showgrid="false"
   inkscape:zoom="0.7055640022"
   inkscape:cx="657.1312544"
   inkscape:cy="257.4602997"
   inkscape:window-x="-8"
   inkscape:window-y="-8"
   inkscape:window-maximized="1"
   inkscape:current-layer="elementy" /><metadata
   id="metadata475"><rdf:RDF><cc:Work
       rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type
         rdf:resource="http://purl.org/dc/dcmitype/StillImage" /></cc:Work></rdf:RDF></metadata><defs
   id="defs473" />












<rect
   transform="matrix(-0.855530,0.517754,0.517754,0.855530,0.000000,0.000000)"
   y="297.9060364"
   x="-554.078064"
   height="226.4941711"
   width="243.5313416"
   id="rect1282"
   style="opacity:0.5;stroke-linejoin:miter;vector-effect:none;stroke-opacity:1;fill-rule:nonzero;fill-opacity:1;paint-order:normal;stroke-dashoffset:0;stroke:none;stroke-linecap:butt;stroke-miterlimit:4;stroke-dasharray:none;stroke-width:1.20990161552;fill:#cccccc" /><rect
   transform="matrix(-0.855530,0.517754,0.517754,0.855530,0.000000,0.000000)"
   y="241.7835846"
   x="-254.4242249"
   height="127.2776947"
   width="123.2689514"
   id="rect1284"
   style="opacity:0.5;stroke-linejoin:miter;vector-effect:none;stroke-opacity:1;fill-rule:nonzero;fill-opacity:1;paint-order:normal;stroke-dashoffset:0;stroke:none;stroke-linecap:butt;stroke-miterlimit:4;stroke-dasharray:none;stroke-width:1.20990161552;fill:#cccccc" /><rect
   transform="matrix(-0.855530,0.517754,0.517754,0.855530,0.000000,0.000000)"
   y="511.3717957"
   x="-180.2624512"
   height="164.3585968"
   width="179.3914032"
   id="rect1286"
   style="opacity:0.5;stroke-linejoin:miter;vector-effect:none;stroke-opacity:1;fill-rule:nonzero;fill-opacity:1;paint-order:normal;stroke-dashoffset:0;stroke:none;stroke-linecap:butt;stroke-miterlimit:4;stroke-dasharray:none;stroke-width:1.20990161552;fill:#cccccc" /></svg>

**Comment by bumbaras** [comment]
I didn't check for response here because was sure there won't be any.
The function to attach files is not working on my end so i can only put a svg code into.
After analyzing applytransform.py code i see You are using functions from simpletransform.py. It clearifies transforms from polygones for example but for rects, and ellipses it leaves me with matrices. How beautiful matrix is and may be, some tools are don't understand matrices at all.
It would be hard to remove matrices from rects because it uses X,Y and width and height and there is no rotate thing in attributes (or maybe there is but i don't know). Rotate is acceptable though.

I can't attach anything here, and can't paste code as well. Probably older firefox prevents this. Just draw in inkspace some rect, rotate it, move it and flip. Inkscape will generate matrix for all three transforms which Your addon leaves intact.

**Comment by Klowner** [comment]
> I didn't check for response here because was sure there won't be any.

... so you just filed a ticket to complain?

**Comment by bumbaras** [comment]
No, You have asked me to give more details in april and I responded after 3 months. I check such portals from time to time. If I expected response i would answered earlier.
For now, i work on files produced by Illustrator CS6 (my friend has to save files for me). Somehow CS6 exporting from AI format is applying transformations permanently. No translate, rotate, scale or matrix. Just plain SVG.
As i mentioned, applytransforms gaves good results for polylines and polygones, but transformed rect or ellipse it can flat only to matrix transformations (which one is the worst i can get). This is probably due to what simpletransform.py can offer. There is computeBBox method although, which seems to manage rect somehow differently than by matrices but don't know if it is suitable to reduce transformations.

What do You mean about "ticket to complain"? Is there some other page I can fill such ticket?

**Comment by bumbaras** [comment]
One more question - inkscape is saving points definition of polygones in format:
points="X1,Y1 X2,Y2 X3,Y3" ... and so on.
Your addon is parsing polygones points considering such notation as reliable.
But when optimized svg in inkscape is choosen as a file format then inkscape removes all commas from points definition - each number in points="" is separated only with space. I don't know if this meets SVG standard or not, but when addon meets such defined polygones it just removes them from drawing.
